packages/design-system/src/components/Table/Table.tsx

```diff
@@ -1,19 +1,70 @@
 import Alert from '../Alert/Alert';
-import PropTypes from 'prop-types';
 import React from 'react';
 import TableCaption from './TableCaption';
 import TableContext from './TableContext';
 import classNames from 'classnames';
 import get from 'lodash/get';
 import uniqueId from 'lodash.uniqueid';
 
-function debounce(fn, ms) {
-  let timer;
-  return () => {
+export interface TableProps {
+  /**
+   * Applies the borderless variation of the table.
+   */
+  borderless?: boolean;
+  /**
+   * The table contents, usually `TableCaption`, `TableHead` and `TableBody`.
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classes to be added to the root table element.
+   */
+  className?: string;
+  /**
+   * Applies the compact variation of the table.
+   */
+  compact?: boolean;
+  /**
+   * @hide-prop [Deprecated] Use compact instead.
+   */
+  dense?: boolean;
+  /**
+   * Applies a horizontal scrollbar and scrollable notice on `TableCaption` when the `Table`'s contents exceed the container width.
+   */
+  scrollable?: boolean;
```

Comment:
Should optional boolean props have a default? This allows you to avoid casting to a boolean in code.